watchtower-plugin/src/net/http.rs

```diff
@@ -119,21 +119,25 @@ pub async fn send_appointment(
 
 /// Generic function to post different types of requests to the tower.
 pub async fn post_request<S: Serialize>(endpoint: &str, data: S) -> Result<Response, RequestError> {
-    reqwest::Client::new()
-        .post(endpoint)
-        .json(&data)
-        .send()
-        .await
-        .map_err(|e| {
-            log::error!("{}", e);
-            if e.is_connect() | e.is_timeout() {
-                RequestError::ConnectionError(
-                    "Cannot connect to the tower. Connection refused".into(),
-                )
-            } else {
-                RequestError::Unexpected("Unexpected error ocurred (see logs for more info)".into())
-            }
-        })
+    let client = if endpoint.contains(".onion:") {
+        let proxy = reqwest::Proxy::http("socks5h://127.0.0.1:9050")
```

Comment:
Can't we ask cln to give us the SOCKS proxy host and port instead?
Or maybe provide them as options with `127.0.0.1:9050` as the default.

Reply:
I'm working on it with the second approach, but it becomes a bit more complex.

Thought about asking cln, but that'd mean that if cln is using `Tor` the plugin always will, and if it is not, the plugin would never. I think we may want to give the user the option to pick any combination.

Reply:
Fixed in [dcbcee3](https://github.com/talaia-labs/rust-teos/pull/96/commits/dcbcee385a45aaf36abf1d823fa6e79ae941325b). Not completely convinced it is the most elegant way though.